src/index.js

```diff
@@ -0,0 +1,204 @@
+        .then(() => this.sls.pluginManager.spawn('ignitor:wrap')),
+
+      'after:invoke:local:invoke': () => bPromise.bind(this)
+        .then(() => this.sls.pluginManager.spawn('ignitor:clean')),
+
+      'before:run:run': () => bPromise.bind(this)
+        .then(() => this.sls.pluginManager.spawn('ignitor:wrap')),
+
+      'after:run:run': () => bPromise.bind(this)
+        .then(() => this.sls.pluginManager.spawn('ignitor:clean')),
+
+      // used when debugging ignitor via command serverless ignitor
+      'ignitor:ignitor': () => bPromise.bind(this)
+        .then(build.prebuild)
+        .then(this.schedule)
+        .then(this.wrap),
+
+      'ignitor:schedule:schedule': () => bPromise.bind(this)
+        .then(this.schedule),
+
+      'ignitor:wrap:wrap': () => bPromise.bind(this)
+        .then(build.prebuild)
+        .then(this.wrap),
+
+      'ignitor:deploy:deploy': () => bPromise.bind(this)
+        .then(this.deploy),
+
+      'ignitor:clean:clean': () => bPromise.bind(this)
+        .then(build.clean),
+    };
+  }
+
+  schedule() {
+    const options = get(this, 'sls.service.custom.ignitor', []);
+    const keys = options.length === 0
+      ? new RegExp('.*', 'g')
+      : new RegExp(
+        options.map((k) => k.replace(/\//g, '')).join('|'),
+        'g',
+      );
+
+    this.service = get(this, 'sls.service.service.name', get(this, 'sls.service.service'));
+    this.stage = get(this, 'optStage', get(this, 'sls.service.provider.stage', '*'));
+    this.region = get(this, 'optRegion', get(this, 'sls.service.provider.region'));
+    this.scheduled = Object.keys(this.sls.service.functions)
+      .filter((name) => name.match(keys));
+
+    if (!this.sls.service.resources) {
+      this.sls.service.resources = { Resources: {} };
+    }
+
+    role.attachRoleToLambda(this.sls.service.functions.ignitorDelegate);
+    role.createLambdaRole(this.sls.service.resources.Resources, {
+      stage: this.stage,
+      service: this.service,
+    });
+  }
+
+  wrap() {
+    this.mapping = {};
+
+    const defaultEvent = {
+      rate: 'rate(5 minutes)',
+      wrapper: 'ignitor.ignitor',
+      input: {
+        ignitor: true,
+      },
+    };
+    const { functions } = this.sls.service;
+    this.scheduled.forEach((name) => {
+      const config = {
+        ...defaultEvent,
+        ...functions[name].ignitor,
+      };
+      const { rate, wrapper, input } = config;
+      if (!this.mapping[rate]) {
+        this.mapping[rate] = [];
+      }
+
+      this.mapping[rate].push({
+        lambda: functions[name].name,
+        input,
+      });
+
+      const { handler } = functions[name];
+      functions[name].handler = build.wrap(name, handler, wrapper, false);
+      log(`Wrapped ${handler}`, JSON.stringify(functions[name], null, 2));
+    });
+
+    // create delegate, and inject configured mapping into the code
+    const delegateCode = delegate.create(this.mapping);
+    build.writeToBuildDir('delegate.js', delegateCode);
```

Comment:
This delegate creation process can probably be migrated to the build util, makes more sense there.